In [1]:
from google.colab import drive
drive.mount('/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /gdrive


In [2]:
%cd /gdrive/My Drive/statistical_learning

/gdrive/My Drive/statistical_learning


In [0]:
import numpy as np
import pandas as pd
from scipy.stats import ttest_1samp, ttest_ind, ttest_rel, t
import matplotlib.pyplot as plt
from math import sqrt

%matplotlib inline

In [4]:
df = pd.read_csv('data.csv')
df.head(3)

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0            1                         57                         62
1            2                        103                        122
2            3                         59                         54

In [5]:
df.describe()


SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
count    30.000000                  30.000000                  30.000000
mean     15.500000                  68.033333                  72.033333
std       8.803408                  20.455980                  24.062395
min       1.000000                  28.000000                  32.000000
25%       8.250000                  54.000000                  55.000000
50%      15.500000                  67.000000                  74.000000
75%      22.750000                  81.500000                  85.750000
max      30.000000                 110.000000                 122.000000

In [8]:
print("Old Scheme Mean:", df['Old Scheme (in thousands)'].mean())
print("New Scheme Mean:", df['New Scheme (in thousands)'].mean())

Old Scheme Mean: 68.03333333333333
New Scheme Mean: 72.03333333333333


In [9]:
alpha = 0.05

scheme1 = df['Old Scheme (in thousands)'] * 1000
scheme2 = df['New Scheme (in thousands)'] * 1000

t_statistic, p_value = ttest_1samp(scheme1 - scheme2, 0)
print(t_statistic, p_value)

t_statistic, p_value = ttest_rel(scheme1, scheme2)
print(t_statistic, p_value)

#since this is a 2 tailed test, need to divide pvalue by 2 to get onetailed result
p_value = p_value / 2
print ("The p value =", p_value)

-1.5559143823544377 0.13057553961337662
-1.5559143823544377 0.13057553961337662
The p value = 0.06528776980668831


Since p-value = 0.06528776980668831 is higher than 0.05, we accept (fail to reject) NULL hypothesis. The new scheme has NOT significantly raised outputs.


**4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:**



a) The probability of a type 1 error? (2.5 points)


**Answer**

Probability of Type I error which is predetermined by significance level. Since the significance level is 0.05, then type I error is 0.05. i.e. 5% probability we make Type I error – rejecting null hypothesis when it is true.

## **b) What is the p- value of the hypothesis test if we test for a difference of $5000? (5 points)**


In [0]:
# H0: μd ≤ 5000  
# HA: μd > 5000

In [12]:
#Number of items
n =  df.shape[0]
print("Count = ", n)

#Standard Deviation
sd = (df['Old Scheme (in thousands)'] - df['New Scheme (in thousands)']).std()
print("Standard deviation = ", sd)

#Degree of Freedom
DF = n-1
print("Degree of Freedom = ", DF)

#With α = 0.05 and df = 29, critical value for t statistic (or t_critical ) will be   1.699127.
t_critical = t.ppf(1-alpha, DF)
print("t_critical = ", t_critical)

# t_critical = (diff_in_mean - uD) / sd / sqrt(n)
# t_critical = (diff_in_mean - 0) / sd / sqrt(n)
diff_in_mean = t_critical / sqrt(n) * sd
print("diff_in_mean = ", diff_in_mean)

#tstat = (d^ - uD) / Sd / sqrt(n)
uD = 5000 / 1000 #5 same unit as dataframe
tstat = (diff_in_mean - uD) /( sd / sqrt(n))
print("tstat = ", tstat)

#calculate the p-value
p_val = t.sf(np.abs(tstat), n-1)
print("p_val = ", p_val)

Count =  30
Standard deviation =  14.08104619937615
Degree of Freedom =  29
t_critical =  1.6991270265334972
diff_in_mean =  4.368176156228719
tstat =  -0.2457659514095499
p_val =  0.4037972524794316


# **C. Power of the test (5 points)**

In [13]:
#Power of the hypothesis test is 1 minus the probability of Type II error. Basically the power of a test is the probability that we make the right decision when the null is not correct (i.e. we correctly reject it) Power of test is 1- beta = 1-0.40379 = 0.5962
beta = p_val
power_of_test = 1 - beta
print("Power of Test = ", round(power_of_test * 100, 2), "%")

Power of Test =  59.62 %
